homeassistant/components/pandora/media_player.py

```diff
@@ -94,35 +94,43 @@ def __init__(self, name: str) -> None:
         self._attr_media_duration = 0
         self._pianobar: pexpect.spawn[str] | None = None
 
-    def turn_on(self) -> None:
-        """Turn the media player on."""
-        if self.state != MediaPlayerState.OFF:
-            return
-        self._pianobar = pexpect.spawn("pianobar", encoding="utf-8")
+    async def _start_pianobar(self) -> bool:
+        pianobar = pexpect.spawn("pianobar", encoding="utf-8")
+        pianobar.delaybeforesend = None
+        # mypy thinks delayafterread must be a float but that is not what pexpect says
+        # https://github.com/pexpect/pexpect/blob/4.9/pexpect/expect.py#L170
+        pianobar.delayafterread = None  # type: ignore[assignment]
```

Comment:
I'm sure that type ignore comments are not typically desired. But I will point out that in addition to the pexpect source being able to handle `None` just fine, the project also explicitly tests `None` values https://github.com/pexpect/pexpect/blob/4.9/tests/test_delay.py#L43, so seems intended and supported behaviour. And any other value, even `0.0`, would reintroduce the (non-aync) sleep that this PR is trying to remove.